routes/xref/lib/search.ts

```diff
@@ -118,34 +121,44 @@ function filter(query: Query, store: Store, options: Options) {
   const cachedValue = cache.get(id);
   if (cachedValue) return cachedValue;
 
-  const byTerm = filterByTerm(query, store);
-  const bySpec = filterBySpec(byTerm, query);
-  const byType = filterByType(bySpec, query);
-  const result = filterByForContext(byType, query, options);
+  let result: DataEntry[] = [];
+  for (const term of getTermVariations(query)) {
+    const byTerm = filterByTerm(term, store);
+    const bySpec = filterBySpec(byTerm, query);
+    const byType = filterByType(bySpec, query);
+    const byForContext = filterByForContext(byType, query, options);
+    if (byForContext.length) {
+      result = byForContext;
+      break;
+    }
+  }
 
   cache.set(id, result);
   return result;
 }
 
-function filterByTerm(query: Query, store: Store) {
+function getTermVariations(query: Query) {
   const { term: inputTerm, types = [] } = query;
 
   const isConcept = types.some(t => CONCEPT_TYPES.has(t));
   const isIDL = types.some(t => IDL_TYPES.has(t));
   const shouldTreatAsConcept = isConcept && !isIDL && !!types.length;
-  let term = shouldTreatAsConcept ? inputTerm.toLowerCase() : inputTerm;
-  if (inputTerm === '""') term = "";
-
-  let termData = store.byTerm[term] || [];
-  if (!termData.length && shouldTreatAsConcept) {
-    for (const altTerm of textVariations(term)) {
-      if (altTerm in store.byTerm) {
-        termData = store.byTerm[altTerm];
-        break;
-      }
-    }
+
+  if (shouldTreatAsConcept) {
+    const term = inputTerm.toLowerCase();
+    return (function* () {
```

Comment:
Nit: can we put the generators as curried functions maybe? Just a bit easier on the eyes.

Reply:
Didn't get it. Can you write an example/suggestion?

Reply:
Something like: 

```JS
return termGenerator(term, {single: shouldTreatAsConcept}) : 

function* termGenerator(term){
   yield term;
   if(!single) yield* textVariations(term); 
}
```

Reply:
Decided over Slack to keep as is.